### ISSUE TITLE
Use Silk.NET.Maths in OpenXR

### ISSUE THREAD

**Issue by Perksey** [dotnet/Silk.NET#506]
We have our own Maths types, but it seems we've generated bindings for OpenXR's maths types and are using them instead.

We should remove our bindings for the OpenXR maths types and use typemaps to use Silk.NET.Maths' types instead.

**Comment by Perksey** [comment]
for info @Exanite not sure if we still want to do this

**Comment by Exanite** [comment]
Yup, I saw this. This is why I have the really long review comment here with a list of potentially relevant types: https://github.com/dotnet/Silk.NET/pull/2590#discussion_r3679747921

**Comment by Exanite** [comment]
I'm leaning towards handling these in some way. Either by adding casts where possible or by directly remapping these types.
Not doing so would be annoying having to copy from a Silk/Numerics type to the corresponding API-specific type.

**Comment by Perksey** [comment]
I'd just remap them tbh, we did this in some places in 2.0 and never heard a complaint. This issue is mostly an oversight in 2.0.
